statsmodels/regression/heckman.py

```diff
@@ -0,0 +1,658 @@
+            **kwargs_mle
+            )
+
+        xbeta_hat = np.asarray(results_mle.params[:num_xvars])  # reg eqn coefs
+        zbeta_hat = np.asarray(results_mle.params[num_xvars:num_xvars+num_zvars])  # selection eqn coefs
+
+        log_sigma_hat = results_mle.params[-2]
+        atanh_rho_hat = results_mle.params[-1]
+
+        sigma_hat = np.exp(log_sigma_hat)
+        rho_hat = np.tanh(atanh_rho_hat)
+
+        scale = results_mle.scale
+
+        xbeta_ncov_hat = results_mle.normalized_cov_params[:num_xvars,:num_xvars]
+        zbeta_ncov_hat = results_mle.normalized_cov_params[
+            num_xvars:(num_xvars+num_zvars),num_xvars:(num_xvars+num_zvars)
+            ]
+
+
+        imr_hat = rho_hat*sigma_hat
+
+        # use the Delta method to compute the variance of lambda (the inverse Mills ratio)
+        log_sigma_var_hat = results_mle.normalized_cov_params[-2,-2] * scale
+        atanh_rho_var_hat = results_mle.normalized_cov_params[-1,-1] * scale
+
+        def grad_lambda(log_sigma, atanh_rho):
+            return np.array([atanh_rho, log_sigma])
+
+        grad_lambda_hat = np.atleast_2d(grad_lambda(sigma_hat, rho_hat))
+        covmat = results_mle.normalized_cov_params[-2:,-2:] * scale
+
+        imr_stderr_hat = np.sqrt(
+            grad_lambda_hat.dot(covmat).dot(grad_lambda_hat.T)[0,0]
+            )
+
+        del grad_lambda_hat
+        del covmat
+
+        # fill in results for this fit, and return
+
+        DUMMY_COEF_STDERR_IMR = 0.
+
+        results = HeckmanResults(self, xbeta_hat,
+            xbeta_ncov_hat, scale,
+            select_res=base.LikelihoodModelResults(None, zbeta_hat, zbeta_ncov_hat, scale),
+            params_inverse_mills=imr_hat, stderr_inverse_mills=imr_stderr_hat,
+            var_reg_error=sigma_hat**2, corr_eqnerrors=rho_hat,
+            method='mle')
+
+        return results
+
+    def loglike(self, params):
+        return self.loglikeobs(params).sum(axis=0)
+
+    def nloglike(self, params):
+        return -self.loglikeobs(params).sum(axis=0)
+
+    def loglikeobs(self, params_all):
+        """
+        Log-likelihood of model.
+
+        Parameters
+        ----------
+        params_all : array-like
+            Parameter estimates, with the parameters for the regression
+            equation coming first, then the parameters for the
+            selection equation, then log sigma, then atanh rho.
+
+        Returns
+        -------
+        loglike : float
+            The value of the log-likelihood function for a Heckman correction model.
+
+        """
+
+        # set up data and parameters needed to compute log likelihood
+        Y, X, Z = self.get_datamats()
+        D = self.treated
+
+        num_xvars = X.shape[1]
+        num_zvars = Z.shape[1]
+
+        xbeta = np.asarray(params_all[:num_xvars])  # reg eqn coefs
+        zbeta = np.asarray(params_all[num_xvars:num_xvars+num_zvars])  # selection eqn coefs
```

Comment:
not good names
which is which?

(I just wanted to quickly check whether binary parameters are before the main model linear parameters)